src/InputStream.js

```diff
@@ -25,7 +37,13 @@ InputStream.prototype = {
     return this.source[this.pos++];
   },
 
-  matchString: function(s) {
+  matchString: function(s, optIgnoreCase) {
+    if (optIgnoreCase) {
+      var substring = this.sourceSlice(this.pos, this.pos + s.length);
+      var match = regExpForString(s, IGNORE_CASE).exec(substring);
```

Comment:
Right, we could cache it in the Apply pexpr. My point was that we can't cache it in the CaseInsensitiveTerminal PExpr.

Do you think I should implement that? Rather than special case CaseInsensitiveTerminal (the Apply.eval() logic is pretty complicated already),  I wonder if we should think about a more general solution for this kind of optimization. It's a bit like constant folding.